ignite/metrics/loss.py

```diff
@@ -1,25 +1,33 @@
 from __future__ import division
 
 from torch.autograd import Variable
-from torch.nn.functional import nll_loss
 
 from .metric import Metric
 from ignite.exceptions import NotComputableError
 
 
-class NegativeLogLikelihood(Metric):
+class Loss(Metric):
+    """
+    Calculates the average loss according to the passed loss_fn.
+
+    `update` must receive output of the form (y_pred, y).
+    """
+    def __init__(self, loss_fn):
+        super(Loss, self).__init__()
+        self._loss_fn = loss_fn
+
     def reset(self):
         self._sum = 0
         self._num_examples = 0
 
     def update(self, output):
         y_pred, y = output
-        loss_sum = nll_loss(Variable(y_pred), Variable(y), size_average=False).data[0]
+        loss_sum = self._loss_fn(Variable(y_pred), Variable(y), size_average=False).data[0]
```

Comment:
should we  pass `volatile=True` into the Variables y_pred

Reply:
yea good catch.